pkg/binary/envoy/runtime.go

```diff
@@ -171,7 +164,11 @@ func (r *Runtime) envoyReady() (int, error) {
 	}
 }
 
-// SendSignal sends a signal to the parent process
-func (r *Runtime) SendSignal(s os.Signal) {
```

Comment:
this was cumulative days to understand, because there are so many signal processing code and it wasn't clear what was for testing and what was needed as barely any comments.

I hope the comment says why it exists at all, until we stop it from existing by moving this type of test to e2e which controls the lifecycle of the process and can kill it directly.